lib/libvarnishapi/vut.c

```diff
@@ -319,15 +323,18 @@ VUT_Fini(struct VUT **vutp)
 	CHECK_OBJ_NOTNULL(vut, VUT_MAGIC);
 	AN(vut->progname);
 
+	errno = 0;
+	if (pfh != NULL && VPF_Remove(pfh) != 0)
+		VUT_Error(vut, 1, "Cannot remove pid file %s: %s", vut->P_arg,
+			  strerror(errno));
+	pfh = NULL;
+
 	free(vut->n_arg);
 	free(vut->P_arg);
 	free(vut->q_arg);
 	free(vut->r_arg);
 	free(vut->t_arg);
 
-	vut_vpf_remove();
-	AZ(pfh);
```

Comment:
This code should be restored, and `pfh` should be NULL even if we failed to remove the PID file. At the very least an error message will have been printed already.